### PR TITLE
Add AtomGPT

### PR DESCRIPTION
**Pull request by knc6**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by knc6** [comment]
This package requires GPU, linux_64_cuda* are passing but not others.

@conda-forge/staged-recipes @conda-forge/help-python PR ready for review

File: recipes/atomgpt/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  commands:
+    - pip check
+    # - train_alignn.py -h
+  requires:
+    - pip
+about:
+  home: https://github.com/usnistgov/atomgpt
+  summary: Atomistic Generative Pretrained Transformer
+  license: NIST-PD
+  license_file: LICENSE
```
Comment (ocefpaf) [inline recipes/atomgpt/meta.yaml:64]:
```suggestion
  license_file: LICENSE.rst
```

File: recipes/atomgpt/LICENSE [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Terms of Use
```
Comment (ocefpaf) [inline recipes/atomgpt/LICENSE:1]: Please remove this one and use the one in the tarball.

File: recipes/atomgpt/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    - wheel
+    - pip
+  run:
+    - python >=3.10
+    - numpy >=1.19.5
+    - scipy >=1.6.1
+    - jarvis-tools >=2021.07.19
+    - pytorch >=2.0.0
+    - mpmath >=1.3.0
+    # - pytorch-cuda >=12.1
```
Comment (ocefpaf) [inline recipes/atomgpt/meta.yaml:30]: Is this an optional dependency? How functional is the package without it?

File: recipes/atomgpt/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    - pydocstyle >=6.0.0
+    - pyparsing >=2.2.1,<3
+
+
+test:
+  imports:
+    - atomgpt
+  commands:
+    - pip check
+    # - train_alignn.py -h
```
Comment (ocefpaf) [inline recipes/atomgpt/meta.yaml:57]: Why is this test commented out?

File: recipes/atomgpt/meta.yaml (hunk 1 above)
Comment (knc6) [reply recipes/atomgpt/meta.yaml]: Update the file and the name

File: recipes/atomgpt/LICENSE (hunk 2 above)
Comment (knc6) [reply recipes/atomgpt/LICENSE]: Resolved

File: recipes/atomgpt/meta.yaml (hunk 3 above)
Comment (knc6) [reply recipes/atomgpt/meta.yaml:30]: these are not optional dependencies

File: recipes/atomgpt/meta.yaml (hunk 4 above)
Comment (knc6) [reply recipes/atomgpt/meta.yaml]: `train_alignn.py` came from another conda recipe template I was using, removed this line now

File: recipes/atomgpt/meta.yaml (hunk 3 above)
Comment (ocefpaf) [reply recipes/atomgpt/meta.yaml:30]: Then commenting it out will break the package.

File: recipes/atomgpt/LICENSE (hunk 2 above)
Comment (ocefpaf) [reply recipes/atomgpt/LICENSE]: You are still adding the file. The file is in the tarball, no need to add it twice.

File: recipes/atomgpt/meta.yaml (hunk 3 above)
Comment (ocefpaf) [reply recipes/atomgpt/meta.yaml:30]: @knc6 is

```
# - pytorch-cuda >=12.1
```

A mandatory or an optional dependency?

Comment (knc6) [reply recipes/atomgpt/meta.yaml:30]: It's optional. I can remove it if needed

Comment (ocefpaf) [reply recipes/atomgpt/meta.yaml:30]: OK. So I can disregard https://github.com/conda-forge/staged-recipes/pull/27357#discussion_r1729186899, right? No need to remove if you plan to re-add it later but please add a comment above on why you commented it out.

Comment (knc6) [reply recipes/atomgpt/meta.yaml:30]: Yes, please disregard that comment. I thought you are talking about jarvis-tools package.

Comment (ocefpaf) [reply recipes/atomgpt/meta.yaml:30]: Can you remove the commented out dependency?

Comment (knc6) [reply recipes/atomgpt/meta.yaml]: Sure, commented it out now.

**Comment by knc6** [comment]
@ocefpaf any additional suggestions?

File: recipes/atomgpt/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    - tqdm >=4.60.0
+    - pandas >=1.2.3
+    - pydantic >=1.8.1
+    - pydantic-settings
+    - flake8 >=3.9.1
+    - pycodestyle >=2.7.0
+    - pydocstyle >=6.0.0
+    - pyparsing >=2.2.1,<3
+
+
```
Comment (ocefpaf) [inline recipes/atomgpt/meta.yaml:50]:
```suggestion
```

File: recipes/atomgpt/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+test:
+  imports:
+    - atomgpt
+  commands:
+    - pip check
+  requires:
+    - pip
+about:
```
Comment (ocefpaf) [inline recipes/atomgpt/meta.yaml:58]:
```suggestion

about:
```

**Comment by ocefpaf** [comment]
> @ocefpaf any additional suggestions?

Just a minor blank line fix. Note that you need to figure out why the solver can't find the proper package combinations there. I can't help you with that.

**Comment by knc6** [comment]
@ocefpaf can we ignore the lint message? this is the best I can get so far.

**Comment by knc6** [comment]
@ocefpaf @xhochy have another look

**Review by xhochy — APPROVED** [review]
Looking good. It might fail to build on the feedstock though. Then you need to switch the docker image to a CUDA one in `conda_build_config.yaml`

**Pull request merged**